### DISCUSSION TITLE
Optimum K for K-means Clustering

### DISCUSSION THREAD

**Discussion by naturallyfunny** [teamalgoritma/community#794] [Q&A] [answered]
Fungsi di bawah ini me-return chart agar kita bisa melihat k optimal dengan metode elbow.
```
fviz_nbclust(
  x = dataframe, 
  FUNcluster = kmeans, 
  method = "wss" 
)
```

### Pertanyaan:
Bagaimana cara mendapatkan k optimal secara otomatis? Misal fungsi yang mereturn k optimal dalam bentuk integer

**Comment by IrfanChairurrachman** [comment, marked as answer]
Dear @ArdiantheGreat ,

Untuk elbow method sendiri penentuannya memang masih sangat subjektif karena melihat penurunan wss yang landai. Namun masih belum terdapat ketentuan komprehensif penurunan yang "landai" secara matematis sehingga belum terdapat fungsi langsung untuk mengeluarkan k optimum dalam bentuk integer.

Kami mempunyai referensi bacaan dan library di python mengenai penentuan k optimum berdasarkan elbow method, bernama autoelbow.
Berikut linknya:
- [dokumentasi library](https://pypi.org/project/autoelbow/#description)
- [artikel penerapan](https://bobrupakroy.medium.com/auto-clustering-32c28b3946f1)
Hanya saja yang patut menjadi catatan adalah masih belum terdapat referensi/artikel yang jelas mengenai bagaimana perhitungan unutk menentukan k optimum dari library di atas. Namun setidaknya bisa menjadi bahan bacaan tambahan bagi Bapak.


Semoga informasi ini dapat membantu. Bila ternyata informasi ini dapat menjawab pertanyaan Anda, Anda dapat “Mark as answer” jawaban ini.

**Reply by naturallyfunny** [reply]
Baik. Saya punya pertanyaan lanjutan mengenai clustering.

Adakah metode/algoritma clustering yang membagikan jumlah cluster secara otomatis?

**Reply by IrfanChairurrachman** [reply]
Untuk algoritma/metode clustering tanpa memasukkan jumlah clustering yang diinginkan salah satunya bisa menggunakan DBSCAN (Density-based spatial clustering of applications with noise). Untuk bacaan dan implementasi mengenai DBScan di R antara lain sebagai berikut:

- [DBSCAN Clustering](https://rpubs.com/Davidlimbong/dbscan)
- [DBSCAN: density-based clustering for discovering clusters in large datasets with noise - Unsupervised Machine Learning](http://www.sthda.com/english/wiki/wiki.php?id_contents=7940)

Semoga informasi ini dapat membantu. Bila ternyata informasi ini dapat menjawab pertanyaan Anda, Anda dapat “Mark as answer” jawaban ini.

**Comment by kinantytasyao** [comment]
Halo @ArdiantheGreat  , karena tidak ada konfirmasi lebih lanjut dan untuk menjaga agar forum diskusi tetap berjalan kondusif, kami izin melakukan lock/close diskusi ini. Semoga dapat dipahami.
